SwanFileBrowser/src/swanfilebrowser.ts

```diff
@@ -0,0 +1,268 @@
+   *
+   * @param path path to the project
+   * @returns json object with the project information.
+   */
+  protected projectInfoRequest(path: string): any {
+    const uri = 'swan/project/info?caller=swanfilebrowser&path=' + path;
+    try {
+      return request<any>(uri, {
+        method: 'GET'
+      });
+    } catch (reason) {
+      console.error(`Error on GET ${uri}.\n${reason}`);
+    }
+  }
+
+  /**
+   * Request to get contents from a path
+   *
+   * @param cwd path to get information from jupyter api
+   * @returns json object with the information of the path
+   */
+  protected contentRequest(cwd: string): any {
+    try {
+      return request<any>('api/contents/' + cwd, {
+        method: 'GET'
+      }).then(rvalue => {
+        return rvalue;
+      });
+    } catch (reason) {
+      console.error(`Error on GET 'api/contents'+ ${cwd}.\n${reason}`);
+    }
+  }
+
+  /**
+   * Method to check if the project information is valid or it is corrupted.
+   *
+   * @param project_data json with project data such as name, stack, release etc.
+   * @returns true if the project is valid or false if the project is corrupted.
+   */
+  protected isValidProject(project_data: JSONObject): boolean {
+    for (const tag in this.project_tags) {
+      if (!(this.project_tags[tag] in project_data)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Overloaded method cd, to check folder information before changing directory.
+   *
+   * @param newValue new path
+   * @returns void promise.
+   */
+  async cd(newValue: string): Promise<void> {
+    if (newValue !== '.') {
+      const content = await this.contentRequest(newValue);
+      if (content.is_project === true) {
+        const project_info = await this.projectInfoRequest(newValue);
+        const project_data = project_info[
+          'project_data'
+        ] as ISwanProjectOptions;
+        if (this.isValidProject(project_data as JSONObject)) {
+          await this.kernelSpecSetPathRequest(newValue);
+          return super.cd(newValue);
+        } else {
+          const okButton = Dialog.okButton({ accept: false });
+          await showErrorMessage(
+            'Project Error:',
+            'Error reading the configuration of project ' +
+              project_data?.name +
+              ', please click OK to define a new one.',
+            [okButton]
+          );
+
+          if (okButton.accept) {
+            await this._commands
+              .execute('swan:edit-project-dialog', {
+                name: project_data?.name,
+                stack: project_data?.stack,
+                release: project_data?.release,
+                platform: project_data?.platform,
+                user_script: project_data?.user_script,
+                corrupted: true
+              })
+              .catch(message => {
+                console.log(message);
+              });
+          }
+          await this.kernelSpecSetPathRequest(this.path);
```

Comment:
Do we need this? I'm not sure we do... I though wether changing the spec manager from another tab would be possible, but Lab doesn't allow you to have more than one window open, right?

If we do need it, we should do it bellow when we also cd to '.'.